tests/test_cli.py

```diff
@@ -595,3 +596,36 @@ def test_cli_with_python_mod():
 def test_doctor():
     res = CliRunner().invoke(doctor, catch_exceptions=False)
     assert res.exit_code == 0
+
+
+def test_mlflow_gc_with_datasets(sqlite_store):
+    store = sqlite_store[0]
+
+    mlflow.set_tracking_uri(sqlite_store[1])
+    mlflow.set_experiment("dataset")
+
+    dataset = numpy_dataset.from_numpy(np.array([1, 2, 3]))
+
+    with mlflow.start_run() as run:
+        experiment_id = run.info.experiment_id
+        mlflow.log_input(dataset)
+
+    experiments = store.search_experiments(filter_string="", view_type=ViewType.ALL)
+
+    # default and datasets
+    assert len(experiments) == 2
+
+    store.delete_experiment(experiment_id)
+
+    # the new experiment is only marked as deleted, not removed
+    experiments = store.search_experiments(filter_string="", view_type=ViewType.ALL)
+    assert len(experiments) == 2
+
+    subprocess.check_output(["mlflow", "gc", "--backend-store-uri", sqlite_store[1]])
```

Comment:
```suggestion
    subprocess.check_call(["mlflow", "gc", "--backend-store-uri", sqlite_store[1]])
```

because we don't use output.